clients/geth/specular/rollup/customerror/customerror.go

```diff
@@ -0,0 +1,48 @@
+package customerror
+
+import (
+	"fmt"
+	"runtime"
+	"strconv"
+)
+
+type CustomError struct {
+	err error
+	pc  uintptr
+}
+
+func Errorf(format string, args ...interface{}) error {
+	pc, _, _, _ := runtime.Caller(1)
+	return &CustomError{
+		err: fmt.Errorf(format, args...),
+		pc:  pc,
+	}
+}
+
+func (ce *CustomError) Error() string {
+	return fmt.Sprintf("%s | %s:%d", ce.err.Error(), getFunctionName(ce.pc), getLineNumber(ce.pc))
+}
```

Comment:
this is a bit strange—now every error is actually 2 errors (one wrapped inside the other), so presumably 2 `unwrap` calls?

i think what we really want is probably just to have our `Errorf` return the original `fmt.Errorf` concatenated with the additional context

Reply:
Errorf is updated